app/models/queue_item.rb

```diff
@@ -0,0 +1,33 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+
+  delegate :category, to: :video
+  delegate :title, to: :video, prefix: :video
+
+  validates_numericality_of :position, {only_integer: true}
+
+  def rating
+    review.rating if review
+  end
+
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.create(user: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
+
+  def category_name 
+    category.name
+  end
+
+  private 
+
+  def review
+    @review ||= Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
Rather than using `where.first` for single record search, in Rails 4+, you should use `find_by`. It is more efficient and correct. Solution videos were recorded in Rails 3.